Sources/Model/ObjC/Common/Core Data/TracksContextManager.m

```diff
@@ -82,12 +93,25 @@ - (NSURL *)applicationSupportURLForAppWithBundleIdentifier:(NSString *)bundleIde
     // available and one should always be able to create a folder in it
     if (error != nil) {
         TracksLogError(@"Failed to create folder for %@ in Application Support: %@, %@", bundleIdentifier, error, [error userInfo]);
-        abort();
+
+        if (self.shouldThrowUponFatalCondition) {
+            @throw [NSException exceptionWithName:TracksApplicationSupportException
+                                           reason:[NSString stringWithFormat:@"Error creating the ApplicationSupport Folder: %@", error]
+                                         userInfo:error.userInfo];
+        } else {
+            abort();
+        }
     }
 
     return folder;
 }
 
+- (BOOL)shouldThrowUponFatalCondition {
+    // We consider it safe to Throw (whenever a Fatal Condition arises) whenever there is no global
+    // NSException handler set
```

Comment:
Whoa, I am no objc exception expert! Can you give me a rundown of how this is working @jleandroperez ? What is `NSGetUncaughtExceptionHandler()`?